code/_core/obj/item/_item_quality.dm

```diff
@@ -10,18 +10,24 @@
 		return 0
 	return FLOOR(clamp( (100 - quality) / (100/5),0,5 ),1)
 
-/obj/item/proc/adjust_quality(var/quality_to_add=0)
-
+/obj/item/proc/can_adjust_quality(var/quality_to_check = quality_max)
 	if(quality == -1)
 		return FALSE
 
-	if(quality >= quality_max) //Cannot add or remove quality.
-		return TRUE
+	if(quality >= quality_to_check) //Cannot add or remove quality.
+		return FALSE
+
+	return TRUE
+
+/obj/item/proc/adjust_quality(var/quality_to_add = 0, var/maximum_quality = quality_max)
+	. = can_adjust_quality(maximum_quality)
+	if(!.)
+		return .
 
 	var/original_quality = quality
 	var/original_damage_num = get_damage_icon_number()
 
-	quality = FLOOR(quality + quality_to_add,0.01)
+	quality = min(FLOOR(quality + quality_to_add,0.01), maximum_quality)
```

Comment:
This is smart. What'd make this perfect is it being  ``max(0,min(FLOOR(quality + quality_to_add,0.01), maximum_quality))`` so it doesn't accidentally go below 0.

Reply:
i think everything that already does adjust_quality down checks for the values afterwards to be safe, but yeah can add some more safety here